tests/test_bookmarks.py

```diff
@@ -226,47 +239,50 @@ def test_run(self):
+                        print("ACTUAL RECORD: {}".format(replicated_created_record))
+                        for key in schema_keys:
+                            val = replicated_created_record[0].get(key)
+                            e_val = created_record.get(key)
+                            if e_val != val:
+                                print("\nDISCREPANCEY | KEY {}: ACTUAL: {} EXPECTED {}".format(key, val, e_val))
+                    # BUG | https://stitchdata.atlassian.net/browse/SRCE-3532
+                    self.assertIn(created_record, second_sync_data,
+                                  msg="Data discrepancy for the created record.")
 
                 # Verify that the updated records are replicated by the 2nd sync and match our expectations
-                if not updated_record in second_sync_data:
-                    print("\n\nDATA DISCREPANCY FOR UPDATED RECORD: {}\n".format(stream))
-                    print("EXPECTED RECORD: {}\n".format(updated_record))
-                    replicated_updated_record = [record for record in second_sync_data
-                                                 if updated_record.get('id') == record.get('id')]
-                    print("ACTUAL RECORD: {}".format(replicated_updated_record[0]))
-                    for key in schema_keys:
-                        val = replicated_updated_record[0].get(key)
-                        e_val = updated_record.get(key)
-                        if e_val != val:
-                            print("\nDISCREPANCEY | KEY {}: ACTUAL: {} EXPECTED {}".format(key, val, e_val))
-                self.assertIn(updated_record, second_sync_data,
-                              msg="Data discrepancy for the updated record.")
+                for updated_record in updated_records.get(stream):
+                    if stream not in self.cannot_update_streams():
+                        if not updated_record in second_sync_data:
+                            print("\n\nDATA DISCREPANCY FOR UPDATED RECORD: {}\n".format(stream))
+                            print("EXPECTED RECORD: {}\n".format(updated_record))
+                            replicated_updated_record = [record for record in second_sync_data
+                                                         if updated_record.get('id') == record.get('id')]
+                            print("ACTUAL RECORD: {}".format(replicated_updated_record[0]))
+                            for key in schema_keys:
+                                val = replicated_updated_record[0].get(key)
+                                e_val = updated_record.get(key)
+                                if e_val != val:
+                                    print("\nDISCREPANCEY | KEY {}: ACTUAL: {} EXPECTED {}".format(key, val, e_val))
+                        self.assertIn(updated_record, second_sync_data,
+                                      msg="Data discrepancy for the updated record.")
```

Comment:
Yeah, see the code I mentioned above again referring to our PR: https://github.com/singer-io/tap-square/pull/25/files#diff-19f36877654ab85cc34581bf5817a1abR190

Reply:
Done.